Neos.Flow/Classes/Session/SessionManager.php

```diff
@@ -11,80 +12,39 @@
  * source code.
  */
 
+use Neos\Cache\Exception as CacheException;
 use Neos\Cache\Exception\NotSupportedByBackendException;
 use Neos\Flow\Annotations as Flow;
 use Neos\Flow\Http\Cookie;
 use Neos\Flow\Session\Data\SessionIdentifier;
 use Neos\Flow\Session\Data\SessionKeyValueStore;
 use Neos\Flow\Session\Data\SessionMetaDataStore;
+use Neos\Flow\Session\Exception\InvalidDataInSessionDataStoreException;
+use Neos\Flow\Session\Exception\SessionNotStartedException;
 use Neos\Flow\Utility\Algorithms;
-use Psr\Log\LoggerInterface;
 
-/**
- * Session Manager
- *
- * @Flow\Scope("singleton")
- */
+#[Flow\Scope("singleton")]
 class SessionManager implements SessionManagerInterface
 {
-    /**
-     * @var SessionInterface
-     */
-    protected $currentSession;
-
-    /**
-     * @var array
-     */
-    protected $remoteSessions;
-
-    /**
-     * Meta data storage for sessions
-     *
-     * @Flow\Inject
-     * @var SessionMetaDataStore
-     */
-    protected $sessionMetaDataStore;
-
-    /**
-     * Storage for sessions data
-     *
-     * @Flow\Inject
-     * @var SessionKeyValueStore
-     */
-    protected $sessionKeyValueStore;
-
-    /**
-     * @var float
-     * @Flow\InjectConfiguration(path="session.garbageCollection.probability")
-     */
-    protected $garbageCollectionProbability;
-
-    /**
-     * @Flow\InjectConfiguration(path="session.garbageCollection.maximumPerRun")
-     * @var integer
-     */
-    protected $garbageCollectionMaximumPerRun;
-
-    /**
-     * @Flow\InjectConfiguration(path="session.inactivityTimeout")
-     * @var integer
-     */
-    protected $inactivityTimeout;
-
-    /**
-     * @Flow\Inject(name="Neos.Flow:SystemLogger")
-     * @var LoggerInterface
-     */
-    protected $logger;
+    protected ?SessionInterface $currentSession = null;
+    protected array $remoteSessions = [];
+
+    public function __construct(
```

Comment:
I took a little extra-mile and reviewed the `InjectConfiguration` support for constructor arguments (https://github.com/neos/flow-development-collection/pull/3086), so we can use that instead of the Objects.yaml-hassle.